pkg/web/apps/apps_service.go

```diff
@@ -26,9 +27,31 @@ func NewService(repository Repository) Service {
 func (a *appsService) GetApps() (*[]models.App, error) {
 	apps, err := a.repository.GetApps()
 
+	// Check for errors and return the appropriate error to the handler
 	if err != nil {
+		if err == models.ErrInternalServerError {
+			return nil, models.ErrInternalServerError
+		}
+
+		return nil, models.ErrNotFound
+	}
+
+	if len(*apps) == 0 {
 		return nil, models.ErrNotFound
 	}
 
+	// iterate over the list of apps and get the deployed versions for each
```

Comment:
All this part is not required if you fetch the data with one query as added in the above comment some example.